homeassistant/components/dynalite/dynalitebase.py

```diff
@@ -0,0 +1,76 @@
+"""Support for the Dynalite devices as entities."""
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DOMAIN, LOGGER
+
+
+def async_setup_entry_base(
+    platform, hass, config_entry, async_add_entities, entity_from_device
+):
+    """Record the async_add_entities function to add them later when received from Dynalite."""
+    LOGGER.debug("Setting up %s entry = %s", platform, config_entry.data)
+    bridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def async_add_entities_platform(devices):
+        # assumes it is called with a single platform
+        added_entities = []
+        for device in devices:
+            if device.category == platform:
+                added_entities.append(entity_from_device(device, bridge))
+        if added_entities:
+            async_add_entities(added_entities)
+
+    bridge.register_add_devices(platform, async_add_entities_platform)
+
+
+class DynaliteBase:  # Deriving from Object so it doesn't override the entity (light, switch, cover, etc.)
```

Comment:
We should inherit from `Entity` since we use the entity api. Python will handle this multiple inheritance case.

Reply:
I want it to inherit entity from the light entity, since I assume Light overrides some of the Entity functions and if I inherit it, it may be overwritten. Any entity that is created (not the base classes) is inheriting from Entity. 
I can change it, but it seemed safer to me, which is why I added the comment. What do you think?

Reply:
It's safe to inherit from `Entity` both here and in the dynalite `Light` entity inheriting class.

Reply:
thanks. done